### PR TITLE
Adding a section in SyncReceiveAndPrefetch.md explaining the faulted state of client & linking doc to main README

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Even after the application disables prefetch in the builder, the `receiveMessage
 IterableStream<ServiceBusReceivedMessage> receiveMessages(int maxMessages, Duration maxWaitTime)
 ```
 
-where `maxMessages` is the maximum number of messages to receive and `maxWaitTime` is the overall duration the SDK should wait for the entire `maxMessages` to arrive.
+where `maxMessages` is the maximum number of messages to receive and `maxWaitTime` is the maximum duration client should wait for entire batch of `maxMessages` messages to arrive.
```
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:13]: > should wait for entire batch of `maxMessages` messages to arrive.
Missing the word 'the':  Should be "should wait for **the** entire batch of `maxMessages` messages to arrive."

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md [hunk 2]
```diff
@@ -63,11 +63,19 @@ Let's take the previous example, say within `maxTimeout` of 5 sec, the service r
 
 If the application decide not to do anything with the messages in the buffer, those are eventually GC-ed. If such messages were requested with PEEK_LOCK mode, then, after the server-side lock duration elapses, a later `receiveMessages` call on another `ServiceBusReceiverClient` object returns the same messages.
 
+## Only one Iterable can be "active"
+
+It is possible to have more than one Iterables to co-exist. For example, 3 Iterable will be allocated and co-exists if the application simply calls `receiveMessages` 3 times in a row.
```
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:68]: Iterable and not Iterables.

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md [hunk 3]
```diff
@@ -79,6 +87,15 @@ The lock timeout configured on the service bus entity and `maxMessages` needs to
 
 Refer to [this][prefetchtradeoff] document for the tradeoff when enabling prefetch (or maxMessages > 1).
 
+## The exception/faulted state in ServiceBusReceiverClient
+
+_The exception topic is not directly related to prefetching, but adding this section given exception/faulted state is also scoped in client object-level (like prefetch-queue) and impacts the behavior of Iterables from `receiveMessages` calls._
+
+The Client types in Service Bus SDK have built-in retry to recover from retriable errors. A `ServiceBusReceiverClient` object can reach a faulted terminal state if it exhausts the maximum retry or encounters a non-retriable error.
+
+Once a `ServiceBusReceiverClient` object runs into such faulted state, an iteration on Iterable returned by `receiveMessages` will throw the relevant exception; this is also expected behavior from any future Iterable objects that the application may obtain from a faulted client object.
```
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:96]: this section is a little verbosey; can this be shortened (or alternately can you include a concrete exception example)?  It basically just says we throw an exception on all future calls once we're in an error state.

**Review by joshfree — APPROVED** [review]
LGTM; left some small comments

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md (hunk 3 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:96]: Yes, I have a task to add a sample for this scenario in our sample directory. I'll shorten the text in this pr and create+link sample as a follow-up.

**Review by ki1729 — APPROVED** [review]

**Pull request merged**